rosbag2_py/test/test_transport.py

```diff
@@ -126,6 +127,89 @@ def test_record_cancel(tmp_path, storage_id):
                     timeout=rclpy.duration.Duration(seconds=3))
 
 
+def test_player_unconfigured():
+    # Test that using a constructor without full configuration raises an error when trying to use
+    # the player
+    player = rosbag2_py.Player()
+    with pytest.raises(RuntimeError):
+        player.play()
+    with pytest.raises(RuntimeError):
+        player.wait_for_playback_to_start()
+    with pytest.raises(RuntimeError):
+        player.wait_for_playback_to_finish()
+    with pytest.raises(RuntimeError):
+        player.stop()
+    with pytest.raises(RuntimeError):
+        player.pause()
+    with pytest.raises(RuntimeError):
+        player.resume()
+    with pytest.raises(RuntimeError):
+        player.play_next()
+    with pytest.raises(RuntimeError):
+        player.burst(1)
+    with pytest.raises(RuntimeError):
+        player.seek(0)
+
+
+@pytest.mark.parametrize('storage_id', TESTED_STORAGE_IDS)
+def test_player_api(storage_id):
+    bag_path = str(RESOURCES_PATH / storage_id / 'talker')
+    assert os.path.exists(bag_path), 'Could not find test bag file: ' + bag_path
+
+    storage_options, _ = get_rosbag_options(bag_path, storage_id)
+
+    play_options = rosbag2_py.PlayOptions()
+    play_options.start_paused = True
+    play_options.topics_to_filter = ['topic']
+
+    player = rosbag2_py.Player(storage_options, play_options, 'info', 'rosbag2_player_test')
+
+    ctx = rclpy.Context()
+    ctx.init()
+    node = rclpy.create_node('test_player_play_pause_resume_seek_stop', context=ctx)
+    msgs = []
+    sub = node.create_subscription(
+        String, '/topic', lambda msg: msgs.append(msg.data), QoSProfile(depth=15))
+    resume_client = node.create_client(Resume, '/rosbag2_player_test/resume')
+    executor = rclpy.executors.SingleThreadedExecutor(context=ctx)
+    executor.add_node(node)
+    executor_thread = threading.Thread(
+        target=executor.spin,
+        daemon=True)
+    executor_thread.start()
+
+    assert wait_for(
+        lambda: sub.get_publisher_count() == 1,
+        timeout=rclpy.duration.Duration(seconds=5),
+    ), sub.get_publisher_count()
+
+    player.play()
+    assert player.wait_for_playback_to_start()
+    assert player.play_next()
+    assert player.play_next()
```

Comment:
```suggestion
    player.play()
    assert player.play_next()
    assert player.play_next()
```
In the test need to make sure that the regular playback scenario works well without calling
`assert player.wait_for_playback_to_start()`

Reply:
Can you explain why it needs to work without calling `wait_for_playback_to_start()`?

Reply:
Though it seems I was wrong. It is better to wait. However, looking into the Player logic - it seems it should work correctly even without waiting to start